File: backend/api/src/main/scala/kpn/api/common/LocationChangesTreeNode.scala
package kpn.api.common

case class LocationChangesTreeNode(
  locationName: String,
  routeChanges: ChangeSetElementRefs, // empty for non-leaf nodes
  nodeChanges: ChangeSetElementRefs, // empty for non-leaf nodes
  children: Seq[LocationChangesTreeNode], // empty for leaf nodes
  happy: Boolean,
  investigate: Boolean
)

File: backend/api/src/main/java/kpn/api/common/LocationChangesTreeNode.java
package kpn.api.common;

import kpn.api.common.ChangeSetElementRefs;

import com.google.common.collect.ImmutableList;

public record LocationChangesTreeNode(
  String locationName,
  ChangeSetElementRefs routeChanges, // empty for non-leaf nodes
  ChangeSetElementRefs nodeChanges, // empty for non-leaf nodes
  ImmutableList<LocationChangesTreeNode> children, // empty for non-leaf nodes
  Boolean happy,
  Boolean investigate
) {
}
